json/src/main/java/io/airlift/json/RecordAutoDetectModule.java

```diff
@@ -20,10 +22,17 @@
 import java.util.stream.Stream;
 
 import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 public class RecordAutoDetectModule
         extends SimpleModule
 {
+    @Deprecated(forRemoval = true)
+    @Retention(RUNTIME)
+    @Target(TYPE)
+    public @interface LegacyRecordIntrospection {}
```

Comment:
Why do we need legacy mode? Upgrading to the new version of airlift requires changing callers to use this annotation, anyway, so they might as well just add/remove the necessary annotations needed for the new serialization behavior.